symbolic-debuginfo/src/macho/bcsymbolmap.rs

```diff
@@ -187,6 +191,194 @@ impl<'a, 'd> Iterator for BcSymbolMapIterator<'a, 'd> {
 
 impl FusedIterator for BcSymbolMapIterator<'_, '_> {}
 
+#[derive(Debug, Error)]
+#[error("{kind}")]
+pub struct PListError {
+    kind: PListErrorKind,
+    #[source]
+    source: Option<Box<dyn Error + Send + Sync + 'static>>,
+}
+
+impl From<elementtree::Error> for PListError {
+    fn from(source: elementtree::Error) -> Self {
+        Self {
+            kind: PListErrorKind::Xml,
+            source: Some(Box::new(source)),
+        }
+    }
+}
+
+impl From<PListErrorKind> for PListError {
+    fn from(kind: PListErrorKind) -> Self {
+        Self { kind, source: None }
+    }
+}
+
+impl From<ParseDebugIdError> for PListError {
+    fn from(source: ParseDebugIdError) -> Self {
+        Self {
+            kind: PListErrorKind::ParseValue,
+            source: Some(Box::new(source)),
+        }
+    }
+}
+
+/// Error kind for [`PListError`].
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PListErrorKind {
```

Comment:
This kind is public, but it is not part of any public signature. Since it is only used for the `Display` implementation, it would be better to make this private.

Same applies to the other kind enumeration in this file.

Reply:
So I finally looked at this and I think the actual mistake is not exposing a `PListError::kind() -> PListErrorKind` method.  See also https://github.com/getsentry/symbolic/pull/357